src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,638 @@
+          let pointerValue =
+            railDomain.min + (railDomain.max - railDomain.min) * percent;
+          pointerValue = roundValueToClosestStep(pointerValue, step);
+
+          const [minVal, maxVal] = getAllowableThumbRange(
+            thumbMode,
+            currentValues,
+            activeMoveIndex,
+            railDomain,
+            step,
+          );
+
+          pointerValue = clamp(pointerValue, minVal, maxVal);
+          const newValues = [...currentValues];
+          newValues[activeMoveIndex] = pointerValue;
+          setCurrentValues(newValues);
+          callback && callback(newValues);
+        }
+      },
+      [activeMoveIndex, railDomain, step, thumbMode, currentValues],
+    );
+
+    const handlePointerMove = React.useCallback(
+      (event: PointerEvent): void => {
+        updateThumbValue(event, onUpdate);
+      },
+      [onUpdate, updateThumbValue],
+    );
+
+    // function called by Thumb keyboard processing
+    const setThumbValue = React.useCallback(
+      (index: number, value: number) => {
+        const newValues = [...currentValues];
+        newValues[index] = value;
+        setCurrentValues(newValues);
+        onChange && onChange(newValues);
+      },
+      [currentValues, onChange],
+    );
+
+    const setThumbMoveIndex = React.useCallback((index: number) => {
+      setActiveMoveIndex(index);
+    }, []);
+
+    const handlePointerUp = React.useCallback(
+      (event: PointerEvent) => {
+        updateThumbValue(event, onChange);
+        setActiveMoveIndex(undefined);
+        event.preventDefault();
+        event.stopPropagation();
+      },
+      [onChange, updateThumbValue],
+    );
+
+    const handlePointerDownOnSlider = React.useCallback(
+      (event: React.PointerEvent) => {
+        if (containerRef.current) {
+          const percent = getPercentageOfRectangle(
+            containerRef.current.getBoundingClientRect(),
+            event.pageX,
+          );
+          let pointerValue =
+            railDomain.min + (railDomain.max - railDomain.min) * percent;
+          pointerValue = roundValueToClosestStep(pointerValue, step);
+
+          const closestValueIndex = getClosestValueIndex(
+            currentValues,
+            pointerValue,
+          );
+          const [minVal, maxVal] = getAllowableThumbRange(
+            thumbMode,
+            currentValues,
+            closestValueIndex,
+            railDomain,
+            step,
+          );
+
+          pointerValue = clamp(pointerValue, minVal, maxVal);
+          const newValues = [...currentValues];
+          newValues[closestValueIndex] = pointerValue;
+          setCurrentValues(newValues);
+          onChange && onChange(newValues);
+        }
+      },
+      [railDomain, step, currentValues, thumbMode, onChange],
+    );
+
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      ownerDoc?.addEventListener('pointermove', handlePointerMove);
```

Comment:
ahh.. I really do not like working with native doc and events in React...
So if some deps change (onUpdate for example), we will create a new event listener without actually removing previous ones and so on..

Reply:
Updated to ensure clean up.